### PR TITLE
JIT: Use compGetTieringName for disassembly output

### PR DESCRIPTION
**Pull request by jakobbotsch**
Also add the code type to the method names, which makes jit-analyze able to differentiate between them.

For example:
```asm
; Assembly listing for method System.Numerics.Tests.Perf_Matrix3x2:MultiplyByMatrixOperatorBenchmark():System.Numerics.Matrix3x2:this (Tier0)
; Emitting BLENDED_CODE for X64 with AVX - Windows
; Tier0 code
; rbp based frame
; partially interruptible
; Final local variable assignments
;
;  V00 this         [V00    ] (  1,  1   )     ref  ->  [rbp+10H]   do-not-enreg[] this class-hnd
;  V01 RetBuf       [V01    ] (  1,  1   )   byref  ->  [rbp+18H]   do-not-enreg[]
;  V02 OutArgs      [V02    ] (  1,  1   )  struct (32) [rsp+00H]   do-not-enreg[XS] addr-exposed "OutgoingArgSpace"
;  V03 tmp1         [V03    ] (  1,  1   )  struct (24) [rbp-18H]   do-not-enreg[HS] hidden-struct-arg "non-inline candidate call"
;  V04 tmp2         [V04    ] (  1,  1   )  struct (24) [rbp-30H]   do-not-enreg[HS] hidden-struct-arg "spilled call-like call argument"
;  V05 tmp3         [V05    ] (  1,  1   )  struct (24) [rbp-48H]   do-not-enreg[XS] addr-exposed "by-value struct argument"
;  V06 tmp4         [V06    ] (  1,  1   )  struct (24) [rbp-60H]   do-not-enreg[XS] addr-exposed "by-value struct argument"
;  V07 tmp5         [V07    ] (  1,  1   )   byref  ->  [rbp-68H]   do-not-enreg[] must-init "argument with side effect"
;
; Lcl frame size = 144
```

Example jit-analyze output:
```scala
Found 1 files with textual diffs.

Summary of Code Size diffs:
(Lower is better)

Total bytes of base: 1179
Total bytes of diff: 1220
Total bytes of delta: 41 (3.48 % of base)
Total relative delta: 0.14
    diff is a regression.
    relative diff is a regression.


Top file regressions (bytes):
          41 : diff.dasm (3.48% of base)

1 total files with Code Size differences (0 improved, 1 regressed), 0 unchanged.

Top method regressions (bytes):
          41 (13.67% of base) : base.dasm - System.Numerics.Tests.Perf_Matrix3x2:MultiplyByMatrixOperatorBenchmark():System.Numerics.Matrix3x2:this (Tier1)

Top method regressions (percentages):
          41 (13.67% of base) : base.dasm - System.Numerics.Tests.Perf_Matrix3x2:MultiplyByMatrixOperatorBenchmark():System.Numerics.Matrix3x2:this (Tier1)

1 total methods with Code Size differences (0 improved, 1 regressed), 3 unchanged.

--------------------------------------------------------------------------------
```

### CODE REVIEW

**Review by EgorBo — APPROVED** [review]

File: src/coreclr/jit/codegencommon.cpp [hunk 1]
```diff
@@ -1785,15 +1785,9 @@ void CodeGen::genGenerateMachineCode()
...
-        else if (compiler->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER1))
-        {
-            printf("; Tier-1 compilation\n");
-        }
-        else if (compiler->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_READYTORUN))
+        printf("; %s code\n", compiler->compGetTieringName(false));
+
+        if (compiler->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_READYTORUN))
         {
             printf("; ReadyToRun compilation\n");
```
[the remark this answers is not in the document]
Comment (jakobbotsch) [reply src/coreclr/jit/codegencommon.cpp]: Sure, done

**Comment by hez2010** [comment]
I'm afraid there will be a breaking change on the compiler explorer side. We may need to update there later to support handling this change as well.

**Comment by jakobbotsch** [comment]
@hez2010 Is it a big problem to fix? Generally the disassembly output is not going to be (very) stable, but I can understand why dependencies could be taken on some of those header fields.

**Comment by hez2010** [comment]
> Is it a big problem to fix?

No. I think it should be quite easy to fix.

**Pull request merged**